runtime/bcutil/ComparingCursor.cpp

```diff
@@ -430,20 +429,27 @@ ComparingCursor::shouldCheckForEquality(DataType dataType, U_32 u32Value)
 
 	switch (dataType) {
 	case SRP_TO_UTF8_CLASS_NAME:
-		if (isComparingLambdaFromSCC()) {
-			/* if the class is a lambda class don't compare the class names because lambda classes might have different index numbers from run to run */
+#if JAVA_SPEC_VERSION < 21
+		if (_context->isLambdaClass()) {
```

Comment:
> ...making readers think that name and fie size comparisons are always ignored for lambda classes

This is the case though, if anything, I think it was a bit misleading before. `_isComparingLambdaFromSCC` [is set to true](https://github.com/eclipse-openj9/openj9/blob/dba9d932e21c71dac4b2bd22846a712cbbeb0217/runtime/bcutil/ROMClassBuilder.cpp#L1475) for every `Lambda` class because `isShareableLambda` [identifies all `Lambda` classes](https://github.com/eclipse-openj9/openj9/blob/dba9d932e21c71dac4b2bd22846a712cbbeb0217/runtime/bcutil/ROMClassBuilder.cpp#L356-L361). The `isShareableLambda` boolean was named as such to discriminate between `Lambda`s (which are all shared in the SCC), and `LambdaForm`s (which currently aren't).

i.e. there isn't a code path where `ComparingCursor` was instantiated with a `Lambda` class  and `_isComparingLambdaFromSCC` is `false`.

Reply:
> there isn't a code path where ComparingCursor was instantiated with a Lambda class and _isComparingLambdaFromSCC is false.

Yes, it is the only case now. There might be new cases added in the future.

The only reason we skip the class name and file size compassion is the special handing when they are in the SCC. If they are not in the SCC, it is incorrect to skip them in the comparison. If you want to change it as it is in this PR, please at least add some comments in the code.

Reply:
Ok, thank you for the feedback. I see two ways forward:
1) Revert the removal of `_isComparingLambdaFromSCC`, or
2) Change the `if` condition to be `_context->isLambdaClass() && _checkRangeInSharedClass`. The latter is set via `romClassIsShared` in the constructor.

Reply:
I am fine with either one.